Interrogator.xUnit/NotConcurrentAttribute.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Interrogator.xUnit.Common;
+using Interrogator.xUnit.Utilities;
+
+namespace Interrogator.xUnit
+{
+	[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+	public class NotConcurrentAttribute : DependsOnAttribute
+	{
+		public enum ConcurrencyScope
+		{
+			Class,
+			Namespace,
+			Assembly
+		}
+
+		public string GroupName { get; }
+		public ConcurrencyScope Scope { get; }
+		public string CurrentMethodName { get; }
```

Comment:
Rather than get this from a parameter (even if it is auto populated), I think I'd prefer the call site getting this attribute to just associate it with the method it got it from.